### ISSUE TITLE
Config.ml version check breaks pinning mirage

### ISSUE THREAD

**Issue by reynir** [mirage/mirage#1565]
I pinned mirage to a local repository (that is, `git+file:///...`) and I get an error because the `dune subst` somehow does not produce a usable version that fits the expected `%u(.%u(.%u))` format:
```
reynir@solsort:~/workspace/qubes-miragevpn$ mirage configure -t xen
couldn't extract version (%u) from e73cf36
reynir@solsort:~/workspace/qubes-miragevpn$ mirage --version
e73cf36
```

**Comment by hannesm** [comment]
hmm, did you run `dune subst` manually? I thought the pieces in lib/functoria/tool.ml:

```OCaml
  if String.equal version ("%%" ^ "VERSION%%") then (
    Log.info (fun m ->
        m "Skipping version check, since our_version is not watermarked");
    Ok ())
```

would hit for pinned mirage versions...

**Comment by reynir** [comment]
No, I did not. I am starting to think it may be another round of this bug in opam https://github.com/ocaml/opam/issues/6145 but I'm unsure.

**Comment by hannesm** [comment]
But of course, if such an error is hit, we can print a warning in the error case instead of failing.

Give me 2 seconds for a PR.

**Comment by hannesm** [comment]
yes, that may be related to the opam issue, are you using 2.2.0 or 2.2.1?

**Comment by hannesm** [comment]
I don't quite understand your last comment. From what I observe there, https://github.com/mirage/mirage/pull/1566 would solve your issues (at the price to not do version checks for mirage pinned to a git version). Is that right? I'm sorry that `git+file:///home/reynir/workspace/mirage#e73cf365` doesn't work with opam, but that may be something to open at the opam issue tracker?

**Comment by reynir** [comment]
Indeed, it seems to be an issue with opam. Yes, #1566 would solve my issue.

**Comment by hannesm** [comment]
fixed by #1566
